contracts/src/libraries/Secp256k1.sol

```diff
@@ -39,17 +39,25 @@ library Secp256k1 {
     // ============================================================
 
     /**
-     * @notice The b coefficient of the secp256k1 curve equation y² = x³ + b.
+     * @notice The 'b' coefficient of the secp256k1 curve equation y² = x³ + ax + b.
+     * @dev For secp256k1, the 'a' coefficient is 0, and 'b' is 7. This is a defining
+     *      parameter of the curve, as specified in standards like "SEC 2: Recommended
+     *      Elliptic Curve Domain Parameters".
      */
     uint256 internal constant B = 7;
 
     /**
-     * @notice The prime field modulus for secp256k1.
+     * @notice The prime field modulus for secp256k1, defining the finite field F_p.
+     * @dev All arithmetic operations on the curve's coordinates are performed modulo this
+     *      prime number. Its value is 2^256 - 2^32 - 977 as specified in standards.
      */
     uint256 internal constant P = 0xfffffffffffffffffffffffffffffffffffffffffffffffffffffffefffffc2f;
 
     /**
-     * @notice The order of the secp256k1 curve.
+     * @notice The order of the generator point of the secp256k1 curve.
+     * @dev This is a large prime number that specifies the number of distinct points
+     *      that can be generated by repeatedly adding the generator point to itself.
+     *      All scalar multiplications are performed modulo N.
```

Comment:
```suggestion
     * @dev All scalar arithmetic are performed modulo N.
```

Nit: I don't think we need to explain the cryptography behind `secp256k1` in the comments. This opens us up to bug bounty submissions of the kind "your cryptography explanation is wrong" etc.